lib/cisco_node_utils/vlan.rb

```diff
@@ -100,19 +100,19 @@ def fabricpath_feature_set(fabricpath_set)
     def mode
       result = config_get('vlan', 'mode', @vlan_id)
       return default_mode if result.nil?
-      result.downcase! if result[/FABRICPATH/]
-      result
+      result[0].downcase! if result[0][/FABRICPATH/]
+      result[0]
     end
 
     def mode=(str)
-      if str.empty?
+      if str[0] == default_mode
         config_set('vlan', 'mode', @vlan_id, 'no', '')
       else
-        if 'fabricpath' == str
+        if 'fabricpath' == str[0]
           fabricpath_feature_set(:enabled) unless
             :enabled == fabricpath_feature
         end
-        config_set('vlan', 'mode', @vlan_id, '', str)
+        config_set('vlan', 'mode', @vlan_id, '', str[0])
       end
     end
```

Comment:
I'm confused about why `str` is now an array? Can you explain? 

The getter returns a string and the minitest that tests this setter passes a string (`v.mode = 'fabricpath'`) . This seems wrong and I'm guessing the minitest won't pass now. Have you run the minitests that validate these APIs ?